src/platform/intel/ace/platform.c

```diff
@@ -115,12 +133,18 @@ static void notify_pm_state_exit(enum pm_state state)
 		rfree(global_imr_ram_storage);
 		global_imr_ram_storage = NULL;
 
-		/* zero the mask before D3 context restore */
-		struct ipc *ipc = ipc_get();
-		ipc->task_mask = 0;
-		ipc->pm_prepare_D3 = false;
+		mem_window_idle_exit();
+		sys_clock_idle_exit();
```

Comment:
I think this should go here: https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/xtensa/intel_adsp/ace/power.c#L293-L317
It's best to keep all things related to HW on the kernel side.